library_test.py

```diff
@@ -0,0 +1,120 @@
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+
+import aiohttp
+
+from midealocal.cloud import get_midea_cloud
+from midealocal.devices import device_selector
+from midealocal.discover import discover
+
+
+def get_arguments() -> tuple[ArgumentParser, Namespace]:
+    """Get parsed passed in arguments."""
+    parser = ArgumentParser(description="midea-local library test")
+    parser.add_argument(
+        "--username",
+        "-u",
+        type=str,
+        help="Set Cloud username",
+    )
+    parser.add_argument("--password", "-p", type=str, help="Set Cloud password")
+    parser.add_argument("--cloud_name", "-cn", type=str, help="Set Cloud name")
+    parser.add_argument(
+        "--configfile",
+        "-cf",
+        type=str,
+        help="Load options from JSON config file. \
+        Command line options override those in the file.",
+    )
+
+    arguments = parser.parse_args()
+    # Re-parse the command line
+    # taking the options in the optional JSON file as a basis
+    if arguments.configfile and Path(arguments.configfile).exists():
+        with Path.open(arguments.configfile, encoding="utf-8") as f:
+            arguments = parser.parse_args(namespace=Namespace(**json.load(f)))
+
+    return parser, arguments
+
+
+async def main() -> None:
+    """Run main."""
+    parser, args = get_arguments()
+
+    if not args.password or not args.username or not args.cloud_name:
+        print("You have to specify all parameters: username, password and Cloud name")
+        parser.print_help()
+        sys.exit(1)
+
+    print("-" * 20)
+    print("Starting network discovery...")
+    devices = discover()
+    print("-" * 20)
+    print("Devices: ", devices)
+    first_device = list(devices.values())[0]
+    print("-" * 20)
+    print("First device: ", first_device)
+    # The device type is in hexadecimal as in midealocal/devices/TYPE
+    type_code = hex(first_device["type"])[2:]
+    print("-" * 20)
+    print("First device type: ", type_code)
+
+    session = aiohttp.ClientSession()
+    cloud = get_midea_cloud(
+        session=session,
+        cloud_name=args.cloud_name,
+        account=args.username,
+        password=args.password,
+    )
+    cloud_keys = await cloud.get_keys(first_device["device_id"])
+    print("-" * 20)
+    print("Fist device Cloud info: ", cloud_keys)
+
+    token = ""
+    key = ""
+    for v in cloud_keys.values():
+        token = v["token"]
+        key = v["key"]
+
+    print("-" * 20)
+    print("Fist device Cloud token: ", token)
+    print("Fist device Cloud key:   ", key)
+
+    # Select the device
+    ac = device_selector(
+        name=type_code,
+        device_id=first_device["device_id"],
+        device_type=first_device["type"],
+        ip_address=first_device["ip_address"],
+        port=first_device["port"],
+        token=token,
+        key=key,
+        protocol=first_device["protocol"],
+        model=first_device["model"],
+        subtype=0,
```

Comment:
Authenticate only works for v3 and this is already handled by connect
```suggestion
    # Connect and authenticate
    ac.connect()
```